pkg/pillar/cmd/client/client.go

```diff
@@ -39,6 +39,8 @@ const (
 	errorTime   = 3 * time.Minute
 	warningTime = 40 * time.Second
 	return400   = false
+	certsRetry  = 3 // try fetch certs 3 times
+	waitDNSCnt  = 3 // allow break out waiting for NetworkStatus update
```

Comment:
Don't need this AFAICT once to change the code to use a timer (instead of a sleep) in the event loop.
Code will then pick up changes (to DNS and to certs) while retrying getUuid and/or selfRegister

Reply:
ok. I'll use the same timer as other two, but break out if we have not got it.